datafusion/physical-expr/src/intervals/interval_aritmetic.rs

```diff
@@ -0,0 +1,533 @@
+            other.lower.clone()
+        } else if other.lower.is_null() {
+            self.lower.clone()
+        } else {
+            max(&self.lower, &other.lower)?
+        };
+        let upper = if self.upper.is_null() {
+            other.upper.clone()
+        } else if other.upper.is_null() {
+            self.upper.clone()
+        } else {
+            min(&self.upper, &other.upper)?
+        };
+        Ok(if !lower.is_null() && !upper.is_null() && lower > upper {
+            // This None value signals an empty interval.
+            None
+        } else {
+            Some(Interval { lower, upper })
+        })
+    }
+
+    // Compute the negation of the interval.
+    #[allow(dead_code)]
+    pub(crate) fn arithmetic_negate(&self) -> Result<Interval> {
+        Ok(Interval {
+            lower: self.upper.arithmetic_negate()?,
+            upper: self.lower.arithmetic_negate()?,
+        })
+    }
+
+    /// Add the given interval (`other`) to this interval. Say we have
+    /// intervals [a1, b1] and [a2, b2], then their sum is [a1 + a2, b1 + b2].
+    /// Note that this represents all possible values the sum can take if
+    /// one can choose single values arbitrarily from each of the operands.
+    pub fn add<T: Borrow<Interval>>(&self, other: T) -> Result<Interval> {
+        let rhs = other.borrow();
+        let lower = if self.lower.is_null() || rhs.lower.is_null() {
+            ScalarValue::try_from(self.lower.get_datatype())
+        } else {
+            self.lower.add(&rhs.lower)
+        }?;
+        let upper = if self.upper.is_null() || rhs.upper.is_null() {
+            ScalarValue::try_from(self.upper.get_datatype())
+        } else {
+            self.upper.add(&rhs.upper)
+        }?;
+        Ok(Interval { lower, upper })
+    }
+
+    /// Subtract the given interval (`other`) from this interval. Say we have
+    /// intervals [a1, b1] and [a2, b2], then their sum is [a1 - b2, b1 - a2].
+    /// Note that this represents all possible values the difference can take
+    /// if one can choose single values arbitrarily from each of the operands.
+    pub fn sub<T: Borrow<Interval>>(&self, other: T) -> Result<Interval> {
+        let rhs = other.borrow();
+        let lower = if self.lower.is_null() || rhs.upper.is_null() {
+            ScalarValue::try_from(self.lower.get_datatype())
+        } else {
+            self.lower.sub(&rhs.upper)
+        }?;
+        let upper = if self.upper.is_null() || rhs.lower.is_null() {
+            ScalarValue::try_from(self.upper.get_datatype())
+        } else {
+            self.upper.sub(&rhs.lower)
+        }?;
+        Ok(Interval { lower, upper })
+    }
+}
+
+/// Indicates whether interval arithmetic is supported for the given operator.
+pub fn is_operator_supported(op: &Operator) -> bool {
+    matches!(
+        op,
+        &Operator::Plus
+            | &Operator::Minus
+            | &Operator::And
+            | &Operator::Gt
```

Comment:
I think `LtEq` and `GtEq` should probably be covered (and would be naturally if we use the existing range analysis)

Reply:
In the short-term, we will add it. We currently implemented an extension on the interval arithmetics library.